cs/src/comm/epoxy-transport/EpoxyTransport.cs

```diff
@@ -199,6 +206,13 @@ public struct TimeoutConfig
             logger = new Logger(logSink, enableDebugLogs);
             // Metrics sink may be null
             metrics = new Metrics(metricsSink);
+
+            shutdownLock = new ReaderWriterLockSlim(LockRecursionPolicy.NoRecursion);
+            isShutdown = false;
+
+            collectionsLock = new object();
```

Comment:
It's a little odd that the shutdownLock is a dedicated *Lock type, but this is just an object. Any reason we'd be worse off doing it consistently?